drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/util/ConstraintValueEditorHelper.java

```diff
@@ -239,7 +240,12 @@ private void isLHSFieldTypeEquivalent(final String boundVariable,
     }
 
     public static boolean isEnumEquivalent(final String[] values, final DropDownData dropDownData) {
-        return Arrays.equals(values, dropDownData.getFixedList());
```

Comment:
This was the root cause, the `dropDownData` was null.

Reply:
@jomarko Do you know why the ```dropDownData``` is ```null``` for ```java.time.LocalDate``` but not, for example, when ```java.util.Date``` (the JIRA states the issue is only evident when using ```LocalDate```)? I have no doubt this PR fixes the symptom but I'd like to know why ```dropDownData``` is ```null``` in the first place.

Reply:
@jomarko I see ```ConstraintValueEditor``` instantiates ```ConstraintValueEditorHelper``` in its constructor and, at this point in time, ```dropDownData``` is guaranteed to always be ```null```. It is only when ```ConstraintValueEditor.refresh()``` is called (after ```ConstraintValueEditorHelper``` has been instantiated) that the dropdown data is actually loaded from the DMO.

IMO we should probably also look at correcting the initialisation order in ```ConstraintValueEditor```.

Reply:
@manstis I added commit according your suggestion.